### PR TITLE
Add integration test cases

### PR DESCRIPTION
**Pull request by ShubhamRasal**
- add integration test cases
- update the go version in `go.mod` to 1.18 for `slice` package

### CODE REVIEW

File: integration_tests/run.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+echo "::group::Build mapcidr"
+rm integration-test mapcidr 2>/dev/null
```
Comment (jdk2588) [inline integration_tests/run.sh]: Add `rm integration-test mapcidr 2>/dev/null` at the end of script.

**Review by jdk2588 — APPROVED** [review]
